src/couch_expiring_cache/src/couch_expiring_cache_server.erl

```diff
@@ -0,0 +1,110 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_expiring_cache_server).
+
+-behaviour(gen_server).
+
+-callback start_link() -> {ok, pid()} | ignore | {error, term()}.
+
+-export([
+    start_link/2
+]).
+
+-export([
+    init/1,
+    terminate/2,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    code_change/3
+]).
+
+
+-define(DEFAULT_BATCH_SIZE, 1000).
+-define(DEFAULT_PERIOD_MSEC, 5000).
+-define(DEFAULT_MAX_JITTER_MSEC, 1000).
+
+
+-include_lib("couch_expiring_cache/include/couch_expiring_cache.hrl").
+
+
+start_link(Name, Opts) when is_atom(Name) ->
+    gen_server:start_link({local, Name}, ?MODULE, Opts#{name => Name}, []).
+
+
+init(Opts) ->
+    ?MODULE = ets:new(?MODULE, [named_table, public, {read_concurrency, true}]),
+    DefaultCacheName = list_to_binary(atom_to_list(maps:get(name, Opts))),
+    Period = maps:get(period, Opts, ?DEFAULT_PERIOD_MSEC),
+    MaxJitter = maps:get(max_jitter, Opts, ?DEFAULT_MAX_JITTER_MSEC),
+    {ok, #{
+        cache_name => maps:get(cache_name, Opts, DefaultCacheName),
+        batch_size => maps:get(batch_size, Opts, ?DEFAULT_BATCH_SIZE),
+        period => Period,
+        max_jitter => MaxJitter,
+        timer_ref => schedule_remove_expired(Period, MaxJitter),
+        oldest_ts => 0,
+        elapsed => 0,
+        largest_elapsed => 0,
```

Comment:
What does `largest_elapsed` track? Is it a stat for debugging?

Reply:
Yeah, since we don't currently have a way to collect metrics, it's just a simple way to see how long it's taking (in the worst case) to delete expired entries. If it gets close to 5 seconds, we might need to increase the period, and/or lower the batch size.